deps/openssl/openssl_common.gypi

```diff
@@ -57,14 +61,15 @@
       ],
     }, {
       # linux and others
-      'cflags': ['-Wno-missing-field-initializers',
-                 ## TODO: check gcc_version>=4.3
-                 '-Wno-old-style-declaration'],
+      'cflags': ['-Wno-missing-field-initializers'],
       'defines': [
         'OPENSSLDIR="/etc/ssl"',
         'ENGINESDIR="/dev/null"',
         'TERMIOS',
       ],
     }],
+    ['is_clang==0 and gcc_version>=43', {
```

Comment:
```suggestion
    ['clang==0', {
```
GCC < 4.8 is not longer supported for any supported node version.

Reply:
@refack I moved `['clang==0', {` inside the the condition check
https://github.com/nodejs/node/blob/62942e9ad7a59b76e9255ea2560bad2245709efc/deps/openssl/openssl_common.gypi#L59 

since from the previous version, the flags `W-no-old-declaration-style` only used for this target. What is your opinion about this?

Reply:
Seems reasonable.
IF you look at the logs for example in https://ci.nodejs.org/job/node-test-commit-linux/25194/nodes=ubuntu1804-64/consoleFull there are no such warnings anyway.